CONTRIBUTING.md

```diff
@@ -0,0 +1,643 @@
+
+### Step 3: Registering the architecture and testing official models
+
+After we are done with detecting whatever parameters are detectable, we can register the architecture and test official models.
+
+Before we register our architecture, we'll add a test using one of the official models. (If your architecture doesn't have accessible official models, you'll have to skip adding the test.)
+
+Create a new function `tests/test_ARCH.py` like this:
+
+```python
+from spandrel.architectures.ARCH import ARCH, load
+
+from .util import (
+    ModelFile,
+    TestImage,
+    assert_image_inference,
+    assert_loads_correctly,
+    disallowed_props,
+)
+
+
+def test_ARCH_load():
+    ...
+
+
+def test_ARCH_model_name(snapshot):
+    file = ModelFile.from_url(
+        "https://example.com/path/to/model_name.pth"
+    )
+    model = file.load_model()
+    assert model == snapshot(exclude=disallowed_props)
+    assert isinstance(model.model, ARCH)
+    assert_image_inference(
+        file,
+        model,
+        [TestImage.SR_16, TestImage.SR_32, TestImage.SR_64],
+    )
+```
+
+Fill in the details as usual and run the test. It should download the model successfully and then fail with an `UnsupportedModelError`. Let's fix this error my registering the architecture.
+
+Open `src\spandrel\__helpers\main_registry.py` and start by importing your architecture in the `from ..architectures import (...)` state. Then get read to scroll all the way down and add a new `ArchSupport` object at the end of `MAIN_REGISTRY.add(...)`. The `ArchSupport` object will tell spandrel how to detect whether an arbitrary state dict (so a state dict of any architecture) is from your architecture. We do this by simply detecting the presence of a particular key. Each architecture has a few keys that are always present in the state dict, and we'll use those for detection.
+
+In the case of DITN, its `ArchSupport` object looks like this:
+
+```python
+ArchSupport(
+    id="DITN",
+    detect=_has_keys(
+        "sft.weight",
+        "UFONE.0.ITLs.0.attn.temperature",
+        "UFONE.0.ITLs.0.ffn.project_in.weight",
+        "UFONE.0.ITLs.0.ffn.dwconv.weight",
+        "UFONE.0.ITLs.0.ffn.project_out.weight",
+        "conv_after_body.weight",
+        "upsample.0.weight",
+    ),
+    load=DITN.load,
+)
+```
+
+Run the test from before again, it should now fail because there's not snapshot for it yet. So let's run the architecture tests from the command line to fix this: `pytest tests/test_ARCH.py --snapshot-update`. Running this, all tests should pass, and a new snapshot should be created. You'll also see a few now images. Those are the snapshots for our inference tests.
```

Comment:
```suggestion
Run the test from before again, it should now fail because there's not snapshot for it yet. So let's run the architecture tests from the command line to fix this: `pytest tests/test_ARCH.py --snapshot-update`. Running this, all tests should pass, and a new snapshot should be created. You'll also see a few new images. Those are the snapshots for our inference tests.
```